### PR TITLE
V7.3.14

### PR DESCRIPTION
**Pull request by mattip**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Build for newly released PyPy 7.3.14

### CODE REVIEW

**Comment by h-vetinari** [comment]
Look what santa brought! :)

On OSX:
```
memprotect.c:16:70: error: use of undeclared identifier 'MAP_ANONYMOUS'
    new_ptr = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
                                                                     ^
1 error generated.
 ```

For that, you'll need to either bump the deployment target, or do something like https://github.com/conda-forge/abseil-cpp-feedstock/blob/main/recipe/patches/0003-add-missing-osx-workaround.patch.

**Comment by mattip** [comment]
I wonder why we don't see this on the buildbot. As far as I can tell (using `xcodebuild -version`) the xcode is up to date there, but `clang --version` shows `Apple clang version 13.1.6` where in the failed build I see `x86_64-apple-darwin13.4.0-clang -O3 -fomit-frame-pointer -mmacosx-version-min=10.9 -arch x86_64 ...`. Confusing.

As for what to do about the failure, what is the conda-forge standard for `-mmacosx-version-min` ? I think CPython is [still on 10.5](https://github.com/python/cpython/blob/f7c5a7a0f9f8186a7fdc8b721468d022fe310b04/Mac/BuildScript/build-installer.py#L155), so I wouldn't want to get too far from that. I guess I will add the patch in your link.

**Comment by h-vetinari** [comment]
> As for what to do about the failure, what is the conda-forge standard for `-mmacosx-version-min`?

Currently 10.9, but will soon™ [switch](https://github.com/conda-forge/conda-forge.github.io/issues/1844) 10.13 (pending infrastructure work).

> I think CPython is [still on 10.5](https://github.com/python/cpython/blob/f7c5a7a0f9f8186a7fdc8b721468d022fe310b04/Mac/BuildScript/build-installer.py#L155), so I wouldn't want to get too far from that.

As described in that issue, I personally think it's utterly pointless to spend dev time on anything below 10.13. Large code bases (chrome, libcxx, +/- all google projects etc.) started dropping support for 10.12 and below, aside from the fact that it has been EOL'd by Apple for over 4 years already.

To the degree that numbers are available, usage numbers[^1] support this (at least IMO), not least because Apple pushes people to upgrade aggressively.

[^1]: modulo the fact that every browser fakes the Macos version past 10.15 in the user agent string, rendering web-based stats after that effectively useless

**Comment by mattip** [comment]
The error is again in darwin:
```
/Users/runner/miniforge3/conda-bld/pypy3.9_1703500807357/work/pypy3/rpython/../pypy/module/_hpy_universal/_vendored/hpy/trace/src/trace_internal.h:64:12: \
    error: call to undeclared function 'clock_gettime'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    return clock_gettime(CLOCK_MONOTONIC_RAW, t);
           ^
```

but `man clock_gettime` on macos says it should be available from `time.h`, and `#include <time.h>` is before this. What is going on?

**Comment by mattip** [comment]
Ahh, the man page also says
```
HISTORY
     These functions first appeared in Mac OSX 10.12
```

So I guess the easiest path will be to update the minimum macos version to 10.13

File: recipe/build.sh [hunk 1]
```diff
@@ -30,6 +30,7 @@ RELEASE_DIR=$PYPY3_SRC_DIR/pypy/tool/release
 PYPY_PKG_NAME=pypy3
 BUILD_DIR=${PREFIX}/../build
 TARGET_DIR=${PREFIX}/../target
+export MACOSX_DEPLOYMENT_TARGET=10.13
```
Comment (h-vetinari) [inline recipe/build.sh:33]: That's not going to help. It needs to be set in `conda_build_config.yaml` and then rerendered

**Comment by mattip** [comment]
@h-vetinari the hint about `conda_build_config.yaml` seems to have done the magic. CI is passing. Would anyone care to review?

**Review by h-vetinari — APPROVED** [review]
LGTM!

**Comment by mattip** [comment]
Hmm. Maybe we should wait for a resolution of pypy/pypy#4816

**Pull request merged**